core/src/Kokkos_CopyViews.hpp

```diff
@@ -871,381 +704,55 @@ struct ViewRemap<DstType, SrcType, ExecSpace, 1> {
       view_copy(exec_space..., dst, src);
     } else {
       p_type ext0(0, std::min(dst.extent(0), src.extent(0)));
-      using sv_adapter_type = CommonSubview<DstType, SrcType, 1, p_type>;
-      sv_adapter_type common_subview(dst, src, ext0);
+      CommonSubview common_subview(dst, src, ext0);
       view_copy(exec_space..., common_subview.dst_sub, common_subview.src_sub);
     }
   }
 };
 
-template <class DstType, class SrcType, class ExecSpace>
-struct ViewRemap<DstType, SrcType, ExecSpace, 2> {
-  using p_type = Kokkos::pair<int64_t, int64_t>;
-
-  template <typename... OptExecSpace>
-  ViewRemap(const DstType& dst, const SrcType& src,
-            const OptExecSpace&... exec_space) {
-    static_assert(
-        sizeof...(OptExecSpace) <= 1,
-        "OptExecSpace must be either empty or be an execution space!");
-
-    if (dst.extent(0) == src.extent(0)) {
-      if (dst.extent(1) == src.extent(1)) {
-        view_copy(exec_space..., dst, src);
-      } else {
-        p_type ext1(0, std::min(dst.extent(1), src.extent(1)));
-        using sv_adapter_type =
-            CommonSubview<DstType, SrcType, 2, Kokkos::ALL_t, p_type>;
-        sv_adapter_type common_subview(dst, src, Kokkos::ALL, ext1);
-        view_copy(exec_space..., common_subview.dst_sub,
-                  common_subview.src_sub);
-      }
-    } else {
-      if (dst.extent(1) == src.extent(1)) {
-        p_type ext0(0, std::min(dst.extent(0), src.extent(0)));
-        using sv_adapter_type =
-            CommonSubview<DstType, SrcType, 2, p_type, Kokkos::ALL_t>;
-        sv_adapter_type common_subview(dst, src, ext0, Kokkos::ALL);
-        view_copy(exec_space..., common_subview.dst_sub,
-                  common_subview.src_sub);
-      } else {
-        p_type ext0(0, std::min(dst.extent(0), src.extent(0)));
-        p_type ext1(0, std::min(dst.extent(1), src.extent(1)));
-        using sv_adapter_type =
-            CommonSubview<DstType, SrcType, 2, p_type, p_type>;
-        sv_adapter_type common_subview(dst, src, ext0, ext1);
-        view_copy(exec_space..., common_subview.dst_sub,
-                  common_subview.src_sub);
-      }
-    }
-  }
-};
-
-template <class DstType, class SrcType, class ExecSpace>
-struct ViewRemap<DstType, SrcType, ExecSpace, 3> {
+template <class DstType, class SrcType, std::size_t... I>
+auto create_common_subview_first_and_last_match(const DstType& dst,
+                                                const SrcType& src,
+                                                std::index_sequence<I...>) {
```

Comment:
Don't we have a requirement that `DstType::rank()` is equal to `SrcType::rank()`, that it is at least 3 (?) and that that it is equal to `sizeof...(I)+2`?
Any particular reason why we are not "deducing" the index sequence and passing it at call site?